src/loggers.jl

```diff
@@ -246,7 +246,10 @@ Sets the record type for the logger.
 * `logger::Logger`: the logger to set.
 * `rec::Record`: A `Record` type to use for logging messages (ie: `DefaultRecord`).
 """
-setrecord!(logger::Logger, rec::Type{R}) where {R<:Record} = logger.record = rec
+function setrecord!(logger::Logger, rec::Type{R}; recursive=false) where {R<:Record}
+    logger.record = rec
+    recursive && setrecord!.(getchildren(logger.name), rec; recursive=true)
+end
```

Comment:
You may want to set the return type. Otherwise this will return `false` when recursive is false and previously this returned `rec`